python/looker_sdk/rtl/api_settings.py

```diff
@@ -40,105 +38,125 @@
     from typing_extensions import Protocol
 
 
-def _convert_bool(val: str, _: bool) -> bool:
-    converted: bool
-    if val.lower() in ("yes", "y", "true", "t", "1"):
-        converted = True
-    elif val.lower() in ("", "no", "n", "false", "f", "0"):
-        converted = False
-    else:
-        raise TypeError
-    return converted
-
-
 class PApiSettings(transport.PTransportSettings, Protocol):
-    def get_client_id(self) -> Optional[str]:
+    looker_url: Optional[str]
+    redirect_uri: Optional[str]
+
+    @property
+    def client_id(self) -> Optional[str]:
         ...
 
-    def get_client_secret(self) -> Optional[str]:
+    @property
+    def client_secret(self) -> Optional[str]:
         ...
 
 
-@attr.s(auto_attribs=True, kw_only=True)
```

Comment:
attrs classes are supposed to be pretty simple data containers. this settings implementation is a bit more than just that so I think it's easier to reason about the code when explicitly typed out (rather than relying on attrs/cattrs)